manager_tests/test_openstack.py

```diff
@@ -0,0 +1,294 @@
+        for node in nodes:
+            if node['id'] not in node_names:
+                break
+            external_id = \
+                node['instances'][0]['runtime_properties'].get('external_id')
+            self.check_openstack_resource(external_id, node['node_type'])
+
+    def check_resources_in_deployment_deleted(self, nodes, node_names):
+
+        for node in nodes:
+            if node['id'] not in node_names:
+                break
+            self.check_openstack_resource(
+                node['instances'][0]['runtime_properties']['external_id'],
+                node['node_type'], exists=False)
+
+    def install_manager(self):
+        self.manager_ip = labs.create_lab()
+
+    def uninstall_manager(self):
+        labs.delete_lab()
+
+    def upload_plugins(self):
+
+        sleep(5)
+        utils.update_plugin_yaml(
+            os.environ['CIRCLE_SHA1'], 'openstack')
+        workspace_path = os.path.join(
+            os.path.abspath('workspace'),
+            'build')
+        sleep(5)
+        utils.upload_plugin(utils.get_wagon_path(workspace_path))
+        sleep(5)
+        for plugin in self.plugins_to_upload:
+            utils.upload_plugin(plugin[0], plugin[1])
+
+    def setUp(self):
+
+        sensitive_data = [
+            os.environ['LAB_PASSWORD'],
+            os.environ['LAB_SERVER'],
+            os.environ['LAB_USERNAME']
+        ]
+        super(TestOpenstack, self).setUp(
+            'openstack.yaml', sensitive_data=sensitive_data)
+
+        os.environ['TEST_APPLICATION_PREFIX'] = \
+            os.environ['CIRCLE_BUILD_NUM']
+
+        if 'ECOSYSTEM_SESSION_MANAGER_IP' not in os.environ:
+            self.install_manager()
+            self.addCleanup(self.uninstall_manager)
+            self.password = 'admin'
+        os.environ['ECOSYSTEM_SESSION_PASSWORD'] = self.password
+        os.environ['ECOSYSTEM_SESSION_MANAGER_IP'] = self.manager_ip
+        utils.initialize_cfy_profile(
+            '{0} -u admin -p {1} -t default_tenant'.format(
+                self.manager_ip, self.password))
+        sleep(300)
+        self.upload_plugins()
+        self.add_os_environment_variables()
+
+    def cleanup_network_deployment(self):
+        # This is just for cleanup.
+        # Because its in the lab, don't care if it passes or fails.
+        utils.execute_command(
+            'cfy uninstall -p ignore_failure=true openstack-example-network')
+
+    def port_fixed_ip_deployment(self):
+        openstack_nodes = [
+            'port'
+        ]
+        utils.execute_command(
+            'cfy blueprints upload blueprints/port-fixed-ip.yaml '
+            '-b fixed-ip')
+        # Create Deployment (Blueprint already uploaded.)
+        if utils.create_deployment(
+                'fixed-ip', inputs={'fixed_ip': '192.168.113.250'}):
+            raise Exception(
+                'Deployment fixed-ip failed.')
+        # Install Deployment.
+        if utils.execute_install('fixed-ip'):
```

Comment:
I think the above operations return type is a little bit confusing because if  `create_deployment` or `execute_install` failed it will return `True`. What come to mind is the always success operation return True

Reply:
It's a good comment. The problem is that it is functioning off of bash return codes, 0=success, >0 = failure